docs/src/main/asciidoc/building-native-image.adoc

```diff
@@ -14,7 +14,61 @@ This guide covers:
 
 This guide takes as input the application developed in the link:getting-started[Getting Started Guide].
 
-== Prerequisites
+== GraalVM
+
+Building a native executable requires using a distribution of GraalVM.
+There are three distributions:
+Oracle GraalVM Community Edition (CE), Oracle GraalVM Enterprise Edition (EE) and Mandrel.
+The differences between the Oracle and Mandrel distributions are as follows:
+
+* Mandrel is a downstream distribution of the Oracle GraalVM CE.
+Mandrel's main goal is to provide a way to build native executables specifically designed to support Quarkus.
+
+* Mandrel releases are built from a code base derived from the upstream Oracle GraalVM CE code base,
+with only minor changes but some significant exclusions that are not necessary for Quarkus native apps.
+They support the same capabilities to build native executables as Oracle GraalVM CE,
+with no significant changes to functionality.
+Notably, they do not include support for polyglot programming.
+The reason for these exclusions is to provide a better level of support for the majority of Quarkus users.
+These exclusions also mean Mandrel offers a considerable reduction in its distribution size
+when compared with Oracle GraalVM CE/EE.
+
+* Mandrel is built slightly differently to Oracle GraalVM CE, using the standard OpenJDK project.
+This means that it does not profit from a few small enhancements that Oracle have added to the version of OpenJDK used to build their own GraalVM downloads.
+This enhancements are omitted because the Red Hat OpenJDK team does not manage them, and cannot vouch for.
+This is particularly important when it comes to comformance and security.
+
+* Mandrel is only available for projects that can be compiled with Java 11.
+If you are using Java 8,
+you should consider using Oracle GraalVM instead.
+The reason for not using Java 8 is because it's a legacy release,
+which most Quarkus users are unlikely to need.
+
+* Mandrel is currently only recommended for building native executables that target Linux containerized environments.
+This means that Mandrel users should use containers to build their native executables.
+If you are building native executables for macOS or Windows target platforms,
+you should consider using Oracle GraalVM instead,
+because Mandrel does not currently target these platforms.
+Building native executables directly on bare metal Linux is possible,
+with details available in the https://github.com/graalvm/mandrel/blob/master/README-Mandrel.md[Mandrel README]
+and https://github.com/graalvm/mandrel/releases[Mandrel releases].
+
+The prerequisites vary slightly depending on whether you are using Oracle GraalVM CE/EE or Mandrel.
+
+== Prerequisites for Mandrel
+
+To complete this guide using Mandrel, you need:
+
+* less than 15 minutes
+* an IDE
+* JDK 11 installed with `JAVA_HOME` configured appropriately
```

Comment:
This seems misleading. The Mandrel release comes with OpenJDK 11 bundled. So it doesn't need OpenJDK 11 just Mandrel, unless somebody wants to build it from source, which I don't think this guide is about. Am I missing something?

Reply:
I wonder if this should be `JAVA_HOME` or `GRAAL_VM_HOME`. I'm thinking the latter pointing to the mandrel install, no?

Reply:
The guide is about using the builder image. So there is nothing to point `GRAALVM_HOME` to.

Reply:
Or to put it more clearly, you won't have mandrel on your system to point `JAVA_HOME` to it. You will only have the `ubi-quarkus-mandrel` image for native compilation part.

Reply:
Apparently it's not clear what this doc is referring to and what those reqs are for. Should we spell it out? I'm guessing the JDK 11 install is for compiling the quarkus app?

Reply:
Not sure there's much we can do here. The sources get compiled by the java version outside the container. If you're on Windows/macOS you can set a JAVA_HOME to Mandrel cos there's no distribution for that. On linux you could download Mandrel and set Java home to that (in the absence of fedora RPMs...etc), but that's as far as you can say. So, with all these caveats I'd probably leave it as is for now.